inbox/auth/oauth.py

```diff
@@ -246,18 +248,15 @@ def authenticate_imap_connection(  # type: ignore[override]
 
             # Raise all IMAP disabled errors except authentication_failed
             # error, which we handle differently.
-            if (
-                isinstance(exc, ImapSupportDisabledError)
-                and exc.reason != "authentication_failed"
-            ):
+            if isinstance(exc, GMailDisabledError | IMAPDisabledError):
                 raise exc from original_exc
 
             log.warning(
                 "Error during IMAP XOAUTH2 login",
                 account_id=account.id,  # type: ignore[attr-defined]
                 error=exc,
             )
-            if not isinstance(exc, ImapSupportDisabledError):
+            if not isinstance(exc, ValidationError):
```

Comment:
The exception `ValidationError` is super confusing to me. It subclasses `AuthError`? Could you add a docstring to it please? Also, maybe we can give it a better name?